utils/xscmanager.go

```diff
@@ -19,3 +26,33 @@ func CreateXscServiceManager(serviceDetails *config.ServerDetails) (*xsc.XscServ
 	}
 	return xsc.New(serviceConfig)
 }
+
+func SendXscLogMessageIfEnabled(errorLog *services.ExternalErrorLog, xscManager *xsc.XscServicesManager) error {
+	if !reportLogErrorEventPossible(xscManager) {
+		return nil
+	}
+	err := xscManager.SendXscLogErrorRequest(errorLog)
+	if err == nil {
+		log.Info("Error successfully reported to Coralogix")
+	}
+	return err
+}
+
+// / Determines if reporting to Coralogix is feasible.
+// Reporting to Coralogix is not feasible under the following conditions: inability to reach the Xsc server, Xsc server is disabled, or Xsc version is below the minimum required version.
+func reportLogErrorEventPossible(xscManager *xsc.XscServicesManager) bool {
+	xscVersion, err := xscManager.GetVersion()
+	if err != nil {
+		log.Warn(fmt.Sprintf("failed to check availability of Xsc service:%s\nReporting to Coralogix is skipped...", err.Error()))
+		return false
+	}
+	if xscVersion == "" {
+		log.Warn("Xsc service is not available. Reporting to Coralogix is skipped...")
+		return false
+	}
+	if err = clientutils.ValidateMinimumVersion(clientutils.Xsc, xscVersion, minXscVersionForErrorReport); err != nil {
+		log.Warn("Xsc version must be 1.7.0 or above in order to use Coralogix report service. Reporting to Coralogix is skipped...")
+		return false
+	}
+	return true
+}
```

Comment:
```suggestion
// Determines if reporting to JFrog analytics is feasible.
func isReportLogErrorEventPossible(xscManager *xsc.XscServicesManager) bool {
	xscVersion, err := xscManager.GetVersion()
	if err != nil {
		log.Debug(fmt.Sprintf("failed to check availability of Xsc service:%s\nReporting to JFrog analytics is skipped...", err.Error()))
		return false
	}
	if xscVersion == "" {
		log.Debug("Xsc service is not available. Reporting to JFrog analytics is skipped...")
		return false
	}
	if err = clientutils.ValidateMinimumVersion(clientutils.Xsc, xscVersion, minXscVersionForErrorReport); err != nil {
		log.Debug(err.Error())
		return false
	}
	return true
}
```

* `ValidateMinimumVersion` already return a formatted error about this, no need to warp in another extra duplicate text.
* Since reporting logs is optional feature, and does not effect the main logic, no need to log in `warn` level, `debug` is OK.